io-engine/src/grpc/v1/replica.rs

```diff
@@ -555,6 +555,18 @@ impl ReplicaRpc for ReplicaService {
                             })
                         }
                     };
+                    // validate snapshot name
+                    if let Some(_r) = UntypedBdev::lookup_by_name(&args.snapshot_name) {
+                        return Err(LvsError::SnapshotNameExist {
```

Comment:
I think you misunderstood my comment. I meant a convertion from source was missing

Reply:
Here, we're missing the map from other Errnos, we're only mapping EIVAL and ENOMEDIUM.
```rust
LvsError::Invalid {
                source, ..
            } => match source {
                Errno::EINVAL => Status::invalid_argument(e.to_string()),
                Errno::ENOMEDIUM => Status::failed_precondition(e.to_string()),
                _ => Status::invalid_argument(e.to_string()),
            },
```

Reply:
Added the mapping for missing Error code used.